bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -158,7 +160,7 @@ public SingleDirectoryDbLedgerStorage(ServerConfiguration conf, LedgerManager le
             throws IOException {
         checkArgument(ledgerDirsManager.getAllLedgerDirs().size() == 1,
                 "Db implementation only allows for one storage dir");
-
+        currentFile = ledgerDirsManager.getAllLedgerDirs().get(0);
```

Comment:
cosmetic change to make the code clearer:

pass `currentFile` to getLedgerDirsListener() and make it a private field in the anonymous class (I also suggest to make it no more anonymous)

I think that we could also pass `gcThread` and make the inner class static, so that it is also easier to write unit tests to it (and I would add them)